### ISSUE TITLE
High number of uniq-repeat-dove reads

### ISSUE THREAD

**Issue by kgun12** [marbl/canu#93]
I'm assembling 7.4Gbp of PacBio data corresponding to a eukaryotic protist with a genome of ~70-110Mbp - this translates to ~67x coverage before trimming.  GC content is rather high at 63%. We've got one assembly using WGS, not done by me, but I found it has a very high amount of duplication that does not always manifest as simple repeats and frequently occurs at the ends of contigs.  When I map several lanes of Illumina TruSeq covaris-sheared data to the assembly, 40% (!) of the reads have non-specific (i.e multiple) alignments - but 96% of the reads do align, so I decided to try an assembly with canu.

Running canu with default parameters, pre-correction I get the following QC:

category                reads       read length
----------------           -------  ----------
middle-missing      10848    17128.20
middle-hump          0            0.00
no-5-prime             0            0.00
no-3-prime             0            0.00

low-coverage        98458      8394.31
unique                   8590       10333.68
repeat-cont            0             0.00
repeat-dove           39122     12758.87

span-repeat            246036    12731.27
uniq-repeat-cont     0               0.00
uniq-repeat-dove    175884    13373.91
uniq-anchor             61            20731.98

The number of uniq-repeat-dove reads seems very high, and given the descriptive text that they "will end contigs, potential to misassemble", I am wondering whether I should tweak the assembly parameters.  Post-correction/trimming, the number of reads drops dramatically from 626,669 to 242,578.  The final assembly seemed to have far less duplicated DNA, as 20% of the Illumina data now maps non-specific, but my N50 is much smaller (~290Kbp vs ~450kbp) - but I'm sure that can be expected given the lack of redundancy now.  Is there anything in particular I can do to increase sensitivity to placing those uniq-repeat-dove reads?  

Thanks!!

**Comment by skoren** [comment]
The pre-correction QC names don't mean much since the reads are corrected before being assembled so unique-repeat-dove won't end assembly, they may just have a repeat on one end so I would't worry about that. The N50 is quite similar (300 vs 450), are the assemblies the same size? One could be larger which would change your N50 denominator.

If you wanted to optimize the assembly, I'd check that you have a good amount of corrected reads (25+X) and that the average length is longer than all your input reads. Since you have a good bit of coverage you can also try corMhapSensitivity=fast which will be more conservative in the overlaps it finds and could separate the repeats better. Finally, you could run a unitigging sweep as suggested in ticket #88 .

**Comment by kgun12** [comment]
Thanks a lot Sergey.  The assembly sizes are pretty similar, but the canu-generated set contains ~900 contigs to the ~600 from celera WGS, so the N50 is a little smaller. I have more confidence in the canu contigs, though, due to the lower amount of duplicated and repetitive DNA in them.

I started a run with corMhapSensitivity=fast and notice that the initial pre-correction QC numbers after mhap.sh are already different: low-coverage was 98458, now 158108; uniq-repeat-dove was 175884, now 11058.  Hopefully that trend will carry through.  

Regarding the unitigging sweep - is the idea there to find the bogart output with the highest reported lenContig ng50 and use the corresponding tig store for the remainder of the pipeline? .. and if so, what's the proper way to execute the remainder of the pipeline with that store?  .. or should I just hard-code the corresponding best ovl and ee parameters to bogart in the stock unitigger.sh script and re-run starting with assembly.

Thanks!!

**Comment by kgun12** [comment]
In addendum, corMhapSensitivity=fast seems to have produced a comparatively worse assembly, with 1,089 contigs (vs 600-700) and N50 of 164kb.  The assembly size is the same.

The unitigger sweep seems to show the best ng50 values at the smallest value of -el ($ovl) of 500 - or at least the smaller values (500,1000,1500) and also smaller -eg and -er of .010 to .030.  I'll check to see how this all is reflected in the non-specific Illumina mapping.

**Comment by JohnUrban** [comment]
When I did the unitigger sweep, I saw similar trends in the top 10 conditions for 2 different assumed genome lengths for NG50 (i.e. 2 top10s):

default params win, not lowcov
corcov_all wins one, corcov_default wins the other
diploid 3x settings dominate top 10 with a couple diploid 2x
ee 0.06 wins one, ee 0.04 wins the other
ee in top10 ranges from 0.025 to 0.065
ee mean = 0.04525
ovl: 500 wins one, 1000 wins the other
ovl ranges from 500 to 3000 in top10
ovl mean = 1300
ovl mode = 1000
ovl: 5 500, 7 1000, 4 1500, 4 2000-3000

This was a pacbio assembly for an insect with a genome in the 300 Mb range, starting with ~36-42X coverage. 

definitions:
lowcov = corMhapSensitivity=high corMinCoverage=2 errorRate=0.035
corcov_all = set corOutCoverage high enough to output all reads
diploid 3x = setting the 2nd repeatdetect parameter in bogart to 3*guessed X coverage

**Comment by StefanoLonardi** [comment]
John, can you explain how to do a "unitigger sweep"? Did you write a script that calls CANU with all these different combination of parameters?

**Comment by JohnUrban** [comment]
Sergey gave an example in #88 . I just adapted from that .

**Comment by JohnUrban** [comment]
RE: Did you write a script that calls CANU with all these different combination of parameters?
No - after the entire Canu assembly was complete, I ran a script in the unitigger directory that continuously called bogart followed by tgStoreDump as in Sergey's example.

**Comment by JohnUrban** [comment]
p.s. as for the lowcov and corOutCoverage settings - I actually did 4 different Canu assemblies (default settings, default+corOutCoverage=80, lowcov settings, lowcov+corOutCoverage=80). Then I ran the loop in the unitigging directory of each of those. Then I was able to compare all results -- and report on the top10 (above) when assuming a given genome size (I always test 2 sizes, one at the bottom and one at the top of the range of estimates).
